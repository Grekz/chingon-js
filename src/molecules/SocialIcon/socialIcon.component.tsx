import * as React from 'react'

import Icon, { IconTypes } from '../../atoms/Icon'

interface ISocialIconProps {
  type: SocialIconType
  size?: string
}

type SocialIconType = 'Facebook' | 'Twitter' | 'LinkedIn' | 'Instagram'

const SocialIcon = ({ type, size }: ISocialIconProps) => (
  <Icon iconComponent={IconTypes[type]} size={size} />
)

export default SocialIcon
