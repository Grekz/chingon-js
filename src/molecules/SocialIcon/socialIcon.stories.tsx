import * as React from 'react'

// tslint:disable-next-line: no-implicit-dependencies
import { storiesOf } from '@storybook/react'

import SocialIcon from '.'

storiesOf('Molecules|Social Icon', module).add('As Default', () => (
  <SocialIcon type="LinkedIn" size="100px" />
))
