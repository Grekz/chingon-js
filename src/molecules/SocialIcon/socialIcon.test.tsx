import { configure, shallow } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import serializer, { matchers } from 'jest-emotion'
import * as React from 'react'
import SocialIcon from '.'

configure({ adapter: new Adapter() })
expect.extend(matchers)
expect.addSnapshotSerializer(serializer)

describe('<List /> rendering', () => {
  it('should render SocialIcon', () => {
    const wrapper = shallow(<SocialIcon type="Facebook" />)
    expect(wrapper).toHaveLength(1)
  })
  it('should match snapshot', () => {
    const tree = shallow(<SocialIcon type="Facebook" />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})
