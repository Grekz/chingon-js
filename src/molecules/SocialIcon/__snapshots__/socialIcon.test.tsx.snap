// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> rendering should match snapshot 1`] = `
<Icon
  iconComponent={[Function]}
/>
`;
